app/routes.py

```diff
@@ -28,15 +29,16 @@ def index():
     else:
         display_widgets = False
         
+    user = current_user.get_ld_user()
+    
+    all_flags = json.dumps(ldclient.get().all_flags(user))
 
-    all_flags = json.dumps(ldclient.get().all_flags(current_user.get_ld_user()))
-
-    beta_features = ldclient.get().variation('dark-theme', current_user.get_ld_user(), False)
+    beta_features = ldclient.get().variation('dark-theme', user, False)
     
     set_theme = '{0}/index.html'.format(current_user.set_path)
 
     return render_template(set_theme, title='Home',
-    display_widgets=display_widgets, all_flags=all_flags, show_beta=beta_features)
+    display_widgets=display_widgets, all_flags=all_flags, show_beta=beta_features, user=user)
```

Comment:
You can get `current_user` directly from jinja in the template. Is there a reason why you are adding this here instead?

Reply:
Yes, i had to do it this way in order to get the experimentation use case to work. From what i could tell current_user would always evaluate to who ever was logged into support service and so when we would call get_ld_user(True) in the experiments route, it wouldnt have the intended effect and i wasnt able to trigger the goal event. 

Once i explicitly called the user in the route and passed it to the JS initialization, it then worked as expected.